### PR TITLE
refactor(nx-plugin): rename autorun to cli executor

### LINKED ISSUE #769: Nx executor shouldn't assume autorun
The `code-pushup` target needs to accept different commands so that the GitHub Action can use it - see https://github.com/code-pushup/github-action/issues/35#issuecomment-2248507862

- [ ] remove `autorun` from executor command - https://github.com/code-pushup/cli/blob/main/packages/nx-plugin/src/executors/autorun/executor.ts#L26
  - default behaviour is the same because `autorun` is implicit, but we can append other commands like `compare` without interference 
- [ ] rename `@code-pushup/nx-plugin:autorun` to `@code-pushup/nx-plugin:cli`

### PR DESCRIPTION
**Pull request by BioPhoton**
**Precondition:**
- #792 

**Blocked until e2e tests are usable:**
- #805
- #808 
- #809 
- #810 

**This PR**

Renames the `autorun` executor to `cli`

It includes:

Closes  #769

### CODE REVIEW

File: packages/nx-plugin/src/executors/cli/schema.json [hunk 1]
```diff
@@ -5,11 +5,10 @@
   "description": "Executes the @code-pushup/cli autorun command See: https://github.com/code-pushup/cli/blob/main/packages/cli/README.md#autorun-command",
   "type": "object",
   "properties": {
-    "project": {
+    "command": {
       "type": "string",
-      "description": "The name of the project.",
+      "description": "The command to run.",
       "x-prompt": "Which project should configure Code Pushup?",
```
Comment (matejchalk) [inline packages/nx-plugin/src/executors/cli/schema.json:11]: I guess the prompt can be removed now that the dropdown's gone?

```suggestion
```

File: packages/nx-plugin/src/executors/internal/types.ts [hunk 2]
```diff
@@ -20,6 +20,13 @@ export type ProjectExecutorOnlyOptions = {
  * CLI types that apply globally for all commands.
  */
 export type GlobalExecutorOptions = {
+  command?:
+    | 'collect'
+    | 'upload'
+    | 'autorun'
+    | 'print-config'
+    | 'compare'
+    | 'history';
```
Comment (matejchalk) [inline packages/nx-plugin/src/executors/internal/types.ts:29]: This list is missing the `merge-diffs` command (used by GitHub Action in monorepo mode).

```suggestion
  command?:
    | 'collect'
    | 'upload'
    | 'autorun'
    | 'print-config'
    | 'compare'
    | 'merge-diffs'
    | 'history';
```

File: packages/nx-plugin/src/plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# @code-pushup/nx-plugin
+
+The Nx Plugin for CodePushup, an open source code quality and conformance tool.
```
Comment (matejchalk) [inline packages/nx-plugin/src/plugin/README.md:3]:
```suggestion
The Nx Plugin for [Code PushUp](https://github.com/code-pushup/cli#readme), an open source code quality and conformance tool.
```

File: packages/nx-plugin/src/plugin/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+```text
+Root/
+├── project-name/
+│   ├── code-pushup.config.ts 👈 generated
+│   └── ...
+└── ...
+```
+
+For details visit the [configuration generator docs](../../src/generators/configuration/README.md).
```
Comment (matejchalk) [inline packages/nx-plugin/src/plugin/README.md:54]: Link can be shortened:

```suggestion
For details visit the [configuration generator docs](../generators/configuration/README.md).
```

File: packages/nx-plugin/src/plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+└── ...
+```
+
+Pass positional arguments to execute a specific command, use named arguments to overwrite defaults.
+
+- `nx run <project-name>:code-pushup --onlyPlugins=eslint`
+- `nx run <project-name>:code-pushup collect`
+- `nx run <project-name>:code-pushup upload --upload.server=https://staging.code-pushup.dev`
+
+For a full list of command visit the [CodePushup CLI documentation](../../../cli/README.md#commands).
```
Comment (matejchalk) [inline packages/nx-plugin/src/plugin/README.md:80]:
```suggestion
For a full list of command visit the [Code PushUp CLI documentation](../../../cli/README.md#commands).
```

File: tools/src/publish/bin/bump-package.ts [hunk 6]
```diff
@@ -26,15 +26,19 @@ try {
   if (version != null) {
     if (packageJson.version === version) {
       console.info(`Package version is already set to ${version}.`);
-      process.exit(0);
+      // process.exit(0);
```
Comment (matejchalk) [inline tools/src/publish/bin/bump-package.ts:29]: Is this temporarily commented out, or should it be removed for good?

Comment (BioPhoton) [inline tools/src/publish/bin/bump-package.ts:29]:
```suggestion
```

Comment (BioPhoton) [reply tools/src/publish/bin/bump-package.ts:29]: removed it. I now update the package.json always to be able to mark it.

File: e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap [hunk 7]
```diff
@@ -18,21 +18,17 @@ Commands:
...
+                      debugging.       [boolean] [default: false]
+      --config       Path to config file. By default it loads code-pushup.config
+                     .(ts|mjs|js).                       [string]
+      --tsconfig     Path to a TypeScript config, to be used when loading config
+                      file.                              [string]
+  -p, --onlyPlugins  List of plugins to run. If not set all plugins are run.
                                                  [array] [default: []]
-  -p, --onlyPlugins     List of plugins to run. If not set all plugins are run.
+  -P, --skipPlugins  List of plugins to skip. If not set all plugins are run.
                                                  [array] [default: []]
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap:32]: I somehow use an older version in the test is seems :( 
No clue really.

[the remark this answers is not in the document]
Comment (matejchalk) [reply e2e/cli-e2e/tests/__snapshots__/help.e2e.test.ts.snap]: Fixed using local `.npmrc` in 4e0c034 :tada:

**Review by matejchalk — APPROVED** [review]

**Pull request merged**